src/backend/distributed/utils/node_metadata.c

```diff
@@ -469,20 +469,25 @@ master_update_node(PG_FUNCTION_ARGS)
 
 	char *newNodeNameString = text_to_cstring(newNodeName);
 
+	List *shardList = NodeShardList(nodeId);
```

Comment:
any interaction with metadata tables should be below `CheckCitusVersion` to prevent crashes